application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubCommand.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Optional;
+import java.util.regex.Matcher;
+
+/**
+ * Slash command (/github) used to search for an issue in one of the repositories listed in the
+ * config. It even comes with auto completion!
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class GitHubCommand extends SlashCommandAdapter {
+    private static final long ONE_MINUTE_IN_MILLIS = 60_000L;
+
+    /**
+     * The command option name of this slash command (this is only there to make the linter happy
+     * lol)
+     */
+    private static final String COMMAND_OPTION_NAME = "title";
+
+    private final GitHubReference reference;
+
+    /**
+     * The last time the cache was updated (millis after Jan 1st 1970)
+     * <p>
+     * The cache updates every minute (see the ONE_MINUTE_IN_MILLIS constant)
+     */
+    private long lastCache;
+    /**
+     * The currently cached auto completion
+     */
+    private List<String> autocompleteCache;
+
+    public GitHubCommand(GitHubReference reference) {
+        super("github", "Search GitHub for an issue", CommandVisibility.GUILD);
+
+        this.reference = reference;
+
+        getData().addOption(OptionType.STRING, COMMAND_OPTION_NAME,
+                "Title of the issue you're looking for", true, true);
+
+        updateCache();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        String title = event.getOption(COMMAND_OPTION_NAME).getAsString();
+        Matcher matcher = GitHubReference.ISSUE_REFERENCE_PATTERN.matcher(title);
+
+        if (!matcher.find()) {
+            event.reply("Could not parse your query").setEphemeral(true).queue();
+
+            return;
+        }
+
+        int id = Integer.parseInt(matcher.group(1));
+        Optional<GHIssue> issue = reference.findIssue(id);
+
+        if (issue.isPresent()) {
+            try {
+                event.replyEmbeds(reference.generateReply(issue.get())).queue();
+            } catch (IOException ex) {
+                throw new UncheckedIOException(ex);
+            }
+        } else {
+            event.reply("Could not find the issue you are looking for").setEphemeral(true).queue();
+        }
+    }
+
+    @Override
+    public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+        String title = event.getOption(COMMAND_OPTION_NAME).getAsString();
+
+        if (title.isEmpty()) {
+            event.replyChoiceStrings(autocompleteCache.stream().limit(25).toList()).queue();
+        } else {
+            event
+                .replyChoiceStrings(autocompleteCache.stream()
+                    .sorted(Comparator.comparingInt(s -> StringDistances.editDistance(title,
+                            s.replaceFirst("\\[\\d+] ", ""))))
+                    .limit(25)
+                    .toList())
+                .queue();
+        }
+
+        if (lastCache <= System.currentTimeMillis() - ONE_MINUTE_IN_MILLIS) {
+            updateCache();
+        }
```

Comment:
u only update after an usage. that means that the usage itself is, in practice, always outdated.

instead, run this on a schedule in the background. use `implements Routine` for this.

Reply:
constant updating is not necessary here. it was never intended for it to update every minute, but instead only once it's really used.

Reply:
but that means that the user always sees an outdated result. the chances for the user to have a good experience are thus close to zero.

Reply:
> u only update after an usage. that means that the usage itself is, in practice, always outdated
Fun part is that, in practice, it's actually not always outdated.

Autocompletion starts when you start typing.

So you type `c`, so this command start updating.

While it's updating you finish your sentence `command rework` or `command`.

And by then it's updated.


Alternatively, we just spam the GitHub API while unnecessary (which I myself wasn't very fond of) 
Or we wait with responding, but that's kinda bad UX, since you want to display option ASAP I'd imagine.

Besides that, most of the time it being outdated won't actually matter. For cases where it does matter, once the user notices the issue isn't being autocompleted, they either simply run the command or they retype it / wait a sec.

I believe this is the better solution myself

Reply:
i was missing the fact that the update runs in the autocompletion. in that case i am okay with it - but still dont really like it. but ur right, in that case its not a real issue.